homeassistant/components/homee/__init__.py

```diff
@@ -0,0 +1,274 @@
+"""The homee integration."""
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from pyHomee import Homee
+from pyHomee.const import AttributeType, NodeProfile
+from pyHomee.model import HomeeAttribute, HomeeNode
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType
+
+from .const import ATTR_HOMEE_DATA, CONF_ADD_HOMEE_DATA, DOMAIN
+from .helpers import get_name_for_enum
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["cover"]
+
+
+@dataclass
+class HomeeRuntimeData:
+    """Homee data class."""
+
+    homee: Homee
+
+
+type HomeeConfigEntry = ConfigEntry[HomeeRuntimeData]
```

Comment:
If this is the only thing you're storing there's no need to wrap it in a dataclass

Reply:
So I do just `entry.runtime_data = homee` to store it?

Reply:
yes

Reply:
But with the typed config entry

Reply:
SO this would make it just `type HomeeConfigEntry = ConfigEntry`

Reply:
with ConfigEntry[Homee]

Reply:
Ah, and access it with entry.homee?

Reply:
no, just `entry.runtime_data`